### PR TITLE
WIP: improve reconfiguration support

### PR DESCRIPTION
**Pull request by aaronkvanmeerten**
Work in progress do not merge, support reconfiguration by timestamps

### CODE REVIEW

File: src/stats_reporter.ts [hunk 1]
```diff
@@ -51,8 +55,16 @@ export default class StatsReporter {
...
     }
 
-    setReconfigureError(status: boolean): void {
+    setReconfigureStart(reconfigureStarted: string): void {
+        // initialize the reconfiguration error handler and start time
+        this.reconfigureStarted = reconfigureStarted;
+        this.reconfigureError = false;
+    }
+
+    setReconfigureEnd(status: boolean): void {
```
Comment (raluca8x8) [inline src/stats_reporter.ts:64]:
```suggestion
    setReconfigureEnd(reconfigureError: boolean): void {
```

File: src/stats_reporter.ts [hunk 2]
```diff
@@ -26,6 +27,8 @@ export default class StatsReporter {
     private shutdownStatus: boolean;
     private shutdownError: boolean;
     private reconfigureError: boolean;
+    private reconfigureComplete: string;
+    private reconfigureStarted: string;
```
Comment (raluca8x8) [inline src/stats_reporter.ts:31]:
```suggestion
    /**
    A string identifying the reconfigure request, representing the timestamp when the autoscaler received the reconfigure request
    */
    private reconfigureStarted: string;
```

File: src/stats_reporter.ts [hunk 3]
```diff
@@ -26,6 +27,8 @@ export default class StatsReporter {
     private shutdownStatus: boolean;
     private shutdownError: boolean;
     private reconfigureError: boolean;
+    private reconfigureComplete: string;
```
Comment (raluca8x8) [inline src/stats_reporter.ts:30]:
```suggestion
    /**
       A string identifying the completed reconfigure request; the reconfigure request identifier was initially received via the reconfigureStarted parameter.
    */
    private reconfigureComplete: string;
```

**Review by raluca8x8 — COMMENTED** [review]
Looks good, I added just some small renaming/comments suggestions

**Review by raluca8x8 — APPROVED** [review]

File: src/app.ts [hunk 4]
```diff
@@ -123,18 +123,22 @@ async function pollForStatus() {
...
         if (pollResult.reconfigure) {
             if (!reconfigureLock) {
+                logger.info('Reconfigure detected, triggering reconfiguration');
                 // set reconfigure lock, only reconfigure if configure is not already running
                 reconfigureLock = true;
                 try {
+                    statsReporter.setReconfigureStart(pollResult.reconfigure);
                     await commandHandler.reconfigure();
-                    statsReporter.setReconfigureError(false);
+                    statsReporter.setReconfigureEnd(false);
```
Comment (oanaianc) [inline src/app.ts:132]: Shouldn't be set to true?  statsReporter.setReconfigureEnd(true);

Comment (aaronkvanmeerten) [reply src/app.ts:132]: No we only set the error to true if we had a problem, in the catch case

**Pull request merged**